### DISCUSSION TITLE
GWP discrepancy between ga and dev versions of api

### DISCUSSION THREAD

**Discussion by pmdirector** [Boavizta/boaviztapi#222] [Q&A] [answered]
If I query the ga version of the api for gwp impacts for, say c5.4xlarge, I get this response for the lifetime manufacturing gwp:
{"impacts":{"gwp":{"manufacture":290.0,"use":460.0,"unit":"kgCO2eq"}
If I repeat the query using the dev version of the api, I get this response:
{"impacts":{"gwp":{"embedded":{"value":202.38,"significant_figures":5,"min":121.77,"max":310.9,"warnings...[no end of life]

Explanation for 290 vs 202?  I would guess embedded > manufacture because dev docs say:
"Embedded impacts include the impacts occurring during raw material extraction, **manufacture**, transport and end of life phases." 

Queries used (verbose on or off makes no difference):
curl -X 'POST'   'https://api.boavizta.org/v1/cloud/aws?instance_type=c5.4xlarge&verbose=true&allocation=TOTAL'   -H 'accept: application/json'   -H 'Content-Type: application/json'
curl -X 'GET' 'http://dev.api.boavizta.org/v1/cloud/instance?provider=aws&instance_type=c5.4xlarge&verbose=true&criteria=gwp&criteria=adp&criteria=pe' -H 'accept: application/json'

Thanks for any help.

**Comment by da-ekchajzer** [comment, marked as answer]
Thanks for your issue.

This difference is due to the evolution in the way we complete data: 

* In the prod version, when we didn't know a piece of data and couldn't complete it based on the information we had, we used a conservative value (that will lead to the maximal impact). 

* In v1, we introduce the notion of min and max impact. When we don't know a data and we can't complete it based on the information we have, we use the average of the data available in our dataset and set a minimum and a maximum data which will be used to evaluate the min and max impacts.

In the case of a c5.4xlarge instance, the difference comes from the ``density'' of the ram. The information is not disclosed by AWS. If you look closely into the verbose : 

### prod 

```json
      "density": {
        "value": 0.625,
        "unit": "GB/cm2",
        "status": "DEFAULT",
        "source": null
      },
```

### dev

```json
"density": {
        "value": 1.2443636363636363,
        "status": "COMPLETED",
        "unit": "GB/cm2",
        "source": "Average of 11 rows",
        "min": 0.625,
        "max": 2.375
      },
```

Hope it solves your issue

**Reply by pmdirector** [reply]
Thank you for the clear and detailed answer.
